xdsl/dialects/math.py

```diff
@@ -487,36 +436,20 @@ class ExpOp(IRDLOperation):
     """
 
     name = "math.exp"
-    fastmath = opt_prop_def(FastMathFlagsAttr)
-    operand = operand_def(AnyFloatConstr)
-    result = result_def(AnyFloatConstr)
 
     traits = traits_def(Pure(), SameOperandsAndResultType())
 
-    def __init__(
-        self, operand: Operation | SSAValue, fastmath: FastMathFlagsAttr | None = None
-    ):
-        attributes = {"fastmath": fastmath}
-
-        operand = SSAValue.get(operand)
-        super().__init__(
-            attributes=attributes, operands=[operand], result_types=[operand.type]
-        )
-
 
 @irdl_op_definition
 class FPowIOp(IRDLOperation):
```

Comment:
Not conforming the base classes. Also the only one that `SameOperandsAndResultType` trait does not apply.